packages/react-api/src/hooks/useTileFeatures.js

```diff
@@ -107,14 +110,16 @@ export default function useTileFeatures({
       setSourceFeaturesReady(false);
 
       debounceIdRef.current = debouncedLoadTiles(tiles);
+      const isDroppingFeatures = tiles?.some(tile => tile.content?.isDroppingFeatures)
+      dispatch(setIsDroppingFeatures({ id: source.id, isDroppingFeatures }))
     },
-    [stopAnyCompute, setSourceFeaturesReady, debouncedLoadTiles, debounceIdRef]
+    [source, stopAnyCompute, setSourceFeaturesReady, debouncedLoadTiles, debounceIdRef, dispatch]
```

Comment:
In the line 30, you have sourceId variable with the `source.id`. Here you are using source as dependency when you're only using its id property, I don't think it's a good practice.
Also, I'd put this logic in loadTiles fn, I don't think it's good to set if it's dropping features before loading the tiles, so both are debounced.